smallrye-reactive-messaging-kafka/src/main/java/io/smallrye/reactive/messaging/kafka/commit/KafkaThrottledLatestProcessedCommit.java

```diff
@@ -105,11 +104,50 @@ private OffsetStore getOffsetStore(TopicPartition topicPartition) {
      */
     @Override
     public void partitionsAssigned(Set<TopicPartition> partitions) {
-        offsetStores.clear();
+        stopFlushAndCheckHealthTimer();
+
+        // Remove all handled partitions that are not in the given list of partitions
```

Comment:
This is something I got wrong. From the documentation:
> The list of partitions that are now assigned to the consumer (previously owned partitions will NOT be included, i.e. this list will only include newly added partitions)

https://kafka.apache.org/24/javadoc/org/apache/kafka/clients/consumer/ConsumerRebalanceListener.html#onPartitionsAssigned-java.util.Collection-

This needs to be used in conjunction with revoked to get the full picture
> partitions - The list of partitions that were assigned to the consumer and now have been reassigned to other consumers (may not include all currently assigned partitions, i.e. there may still be some partitions left)

https://kafka.apache.org/24/javadoc/org/apache/kafka/clients/consumer/ConsumerRebalanceListener.html#onPartitionsRevoked-java.util.Collection-


If the idea is to remove partitions from the store then that should be done in partitionsRevoked.

Reply:
Hum... ok, I should not remove the partitions.
However, before we were _Clearing_ the full store was wrong. 

The problem is that in some cases I received records BEFORE this callback is called. This is because of the scheduling in the event loop.

Reply:
Yeah, it was wrong.

> This is something I got wrong

Reply:
> The problem is that in some cases I received records BEFORE this callback is called. 

Perhaps we should use the pause/fetch strategy we use for the user supplied consumer re-balance as well. In this case we'd have to return a Uni.

Reply:
That would not always work, I've seen this issue when the application starts, where we are not in a pause/resume cycle. 

Let's revisit this when we have the blocking callbacks. We are trying to work around an issue that should not be an issue...